src/connector/src/parser/debezium/mongo_json_parser.rs

```diff
@@ -0,0 +1,261 @@
+    let bson_str = value.as_str().unwrap_or_default();
+    let bson_value: bson::Bson = serde_json::from_str::<serde_json::Value>(bson_str)?.try_into()?;
+    let bson_doc = bson_value.as_document().ok_or_else(|| {
+        RwError::from(ProtocolError(
+            "Debezuim Mongo requires payload is a document".into(),
+        ))
+    })?;
+    let id_field = bson_doc
+        .get("id")
+        .ok_or_else(|| {
+            RwError::from(ProtocolError(
+                "Debezuim Mongo requires document has a `_id` field".into(),
+            ))
+        })?
+        .clone();
+    let id: Datum = match id_type {
+        DataType::Jsonb => ScalarImpl::Jsonb(id_field.into_relaxed_extjson().into()).into(),
+        DataType::Varchar => match id_field {
+            bson::Bson::String(s) => Some(ScalarImpl::Utf8(s.into())),
+            bson::Bson::ObjectId(oid) => Some(ScalarImpl::Utf8(oid.to_string().into())),
+            _ => Err(RwError::from(ProtocolError(format!(
+                "Can not convert bson {:?} to {:?}",
+                id_field.element_type(),
+                id_type
+            ))))?,
+        },
+        DataType::Int32 | DataType::Int64 => match id_field {
+            bson::Bson::Int32(i) => Some(ScalarImpl::Int32(i)),
+            bson::Bson::Int64(i) => Some(ScalarImpl::Int64(i)),
+            _ => Err(RwError::from(ProtocolError(format!(
+                "Can not convert bson {:?} to {:?}",
+                id_field.element_type(),
+                id_type
+            ))))?,
+        },
+        _ => unreachable!("DebeziumMongoJsonParser::new must ensure _id column datatyps."),
+    };
+    let payload: Datum = match payload_type {
+        DataType::Jsonb => ScalarImpl::Jsonb(bson_value.into_relaxed_extjson().into()).into(),
+        DataType::Varchar => ScalarImpl::Utf8(bson_str.into()).into(),
+        _ => unreachable!("DebeziumMongoJsonParser::new must ensure payload column datatyps."),
+    };
+
+    Ok((id, payload))
+}
+impl DebeziumMongoJsonParser {
+    pub fn new(rw_columns: Vec<SourceColumnDesc>, source_ctx: SourceContextRef) -> Result<Self> {
+        let id_column = rw_columns
+            .iter()
+            .find(|desc| {
+                desc.name == "_id"
+                    && matches!(
+                        desc.data_type,
+                        DataType::Jsonb
+                            | DataType::Varchar
+                            | DataType::Int32
+                            | DataType::Int64
+                    )
+            })
+            .ok_or_else(|| RwError::from(ProtocolError(
+                "Debezuim Mongo needs a `_id` column with supported types (Varchar Jsonb int32 int64) in table".into(),
+            )))?.clone();
+        let payload_column = rw_columns
+            .iter()
+            .find(|desc| {
+                desc.name == "payload"
+                    && matches!(desc.data_type, DataType::Jsonb | DataType::Varchar)
```

Comment:
in which case will it be a varchar

Reply:
Here we transform the MongoDB extended JSON format from 'Canonical Mode' to 'Relaxed Mode' and there is no '$numberLong' anymore. However, we will retain the payload's string type to ensure that the original payload can still be accessed if necessary.